src/wayland/mod.rs

```diff
@@ -17,17 +20,31 @@ use buffer::WaylandBuffer;
 
 struct State;
 
-pub struct WaylandDisplayImpl {
-    conn: Connection,
+pub struct WaylandDisplayImpl<D: ?Sized> {
+    conn: Option<Connection>,
```

Comment:
Instead of using an `Option`, I'd probably just rely on field order, and have a safety comment indicating this requirement.

There's also `ManuallyDrop`, though the docs for it seem to just recommend using field order for this sort of thing instead of dealing with it: https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html#manuallydrop-and-drop-order

Reply:
I'd rather rely on `Option` than unsafe `ManuallyDrop` or fragile field ordering.